src/mozillavpn.cpp

```diff
@@ -423,6 +423,11 @@ void MozillaVPN::maybeStateMain() {
 #endif
 }
 
+void MozillaVPN::setEntryServerPublicKey(const QString& publicKey) {
+  logger.debug() << "Set entry-server public key:" << logger.keys(publicKey);
+  m_entryServerPublicKey = publicKey;
+}
+
 void MozillaVPN::setServerPublicKey(const QString& publicKey) {
   logger.debug() << "Set server public key:" << logger.keys(publicKey);
   m_serverPublicKey = publicKey;
```

Comment:
Though it requires a broader bit of refactoring, we should rename `m_serverPublicKey` to `m_exitServerPublicKey` to maintain consistency between these two and to make it clear which of the two servers we are referring to. In the case of single-hop, the entry and exit servers should be set to the same value.

This should eliminate a lot of `vpn->multiphop() ? entry : exit` kind of checks.